File: src/main/resources/Fibonacci.rb
puts "Enter the value of N to find the Nth Fibonacci number:"
n = gets.chomp.to_i

if n < 0
  puts "Fibonacci numbers are not defined for negative indices."
else
  a = 0
  b = 1
  i = 2

  if n == 0
    fib = a
  elsif n == 1
    fib = b
  else
    while i <= n
      fib = a + b
      a = b
      b = fib
      i = i + 1
    end
  end

  puts "The #{n}th Fibonacci number is: #{fib}"
end

File: src/main/resources/Fibonacci.kt
fun main() {
    print("Enter the value of N to find the Nth Fibonacci number: ")
    val n = readln().toInt()

    if (n <= 0) {
        println("Please enter a positive integer.")
    } else {
        var a = 0
        var b = 1
        var fib = 0

        if (n == 1) {
            fib = a
        } else if (n == 2) {
            fib = b
        } else {
            var i = 3
            while (i <= n) {
                fib = a + b
                a = b
                b = fib
                i++
            }
        }

        println("The ${n}th Fibonacci number is: $fib")
    }
}